.gitignore

```diff
@@ -166,9 +166,8 @@ numpy/core/src/umath/simd.inc
 numpy/core/src/umath/struct_ufunc_test.c
 numpy/core/src/umath/test_rational.c
 numpy/core/src/umath/umath_tests.c
-numpy/core/src/umath/_umath_tests.dispatch.avx2.c
-numpy/core/src/umath/_umath_tests.dispatch.h
-numpy/core/src/umath/_umath_tests.dispatch.sse41.c
+numpy/core/src/*/*.dispatch.*.c
+numpy/core/src/*/*.dispatch.h
```

Comment:
@seiko2plus shouldn't we be generating files in build/* rather than in the source directories, or is this similar to cython-generated files?

Reply:
yes, should be generated into build/src.*, but maybe @Qiyu8 use `--inplace`.

Reply:
yes, I used `-i` option during build.

Reply:
I guess you used inplace option `-i` because you received error like that
```Bash
: fatal error: einsum.dispatch.h: No such file or directory
```
well it's my bad, `CcompilerOpt` should include `build/src.*/numpy/core/src/multiarray` to the search dir, even if it wasn't added by 'core/setup.py'.

This issue should be fixed by #16858.

Reply:
#16858 is merged.

Reply:
thank you @mattip.

Reply:
I think we should keep this ignore part, I build the source with `-i` option, and the `*.dispatch.*.c` will generated.

Reply:
By default people should not be using `-i` since it messes up the source directory, as you discovered. Why don't you want to use the build directory? Is there another problem that using `-i` fixes?

Reply:
build directory is good , no problem if I using `-i` option. will remove from ignore list.